### DISCUSSION TITLE
Add EMA (slow and fast) values to MACD results

### DISCUSSION THREAD

**Discussion by sophisma** [DaveSkender/Stock.Indicators#593] [Ideas]
Hi, like the topic says, is it possible to add the EMAs to the MACD results?

**Comment by DaveSkender** [comment]
Yes, I think so.  You're referring to the underlying EMAs, right?
I'd probably add it to an extended analysis version of the method (see #320), to keep the base version simpler -- not sure yet, since these are already generated in the base version.

**Reply by sophisma** [reply]
Yes, the underlying ones.
Not sure if it takes a huge performance hit, only by benchmarking it with a substantial dataset will it be possible to evaluate.
It could be interesting to add these intermediate values to other responses too, for instance with the KAMA it would be nice to have the full picture of what happens.
![image](https://user-images.githubusercontent.com/2579705/138165059-009fd019-fcd4-4e74-99e5-2b0dcd790837.png)
I know it will clutter the response but internally the values are already being calculated. 
Maintaining multiple versions comes with its challenges too but I think you're in better position to know what's the best way.

**Reply by DaveSkender** [reply]
Technically not hard to do; I'm just trying to balance the interest of different user types.  Some prefer a more minimalist approach.

**Reply by DaveSkender** [reply]
I've added this to the backlog - see #594 for tracking

**Reply by DaveSkender** [reply]
I think for KAMA, we already provide the Efficiency Ratio (ER).  The other values don't add much value IMO since they're all correlated.  My general position is that if there's a good use case to provide some of the interim data, then I'd consider adding it.  If you're just curious about what's happening in the algorithm, fork the repo and goof around with it!  We're open source!  :smiley:

**Comment by DaveSkender** [comment]
These MACD EMAs will be available in the next version after 1.18.0
Not sure when the next release will be, but it's merged.

**Reply by sophisma** [reply]
Tank you very much.
That was quick.
